source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -605,22 +621,30 @@ const std::string& TunnelResponseTrailers::key() {
 }
 
 TunnelingConfigHelperImpl::TunnelingConfigHelperImpl(
-    const envoy::extensions::filters::network::tcp_proxy::v3::TcpProxy_TunnelingConfig&
-        config_message,
+    Stats::Scope& stats_scope,
+    const envoy::extensions::filters::network::tcp_proxy::v3::TcpProxy& config_message,
     Server::Configuration::FactoryContext& context)
-    : use_post_(config_message.use_post()),
-      header_parser_(Envoy::Router::HeaderParser::configure(config_message.headers_to_add())),
-      propagate_response_headers_(config_message.propagate_response_headers()),
-      propagate_response_trailers_(config_message.propagate_response_trailers()),
-      post_path_(config_message.post_path()) {
+    : use_post_(config_message.tunneling_config().use_post()),
+      header_parser_(Envoy::Router::HeaderParser::configure(
+          config_message.tunneling_config().headers_to_add())),
+      propagate_response_headers_(config_message.tunneling_config().propagate_response_headers()),
+      propagate_response_trailers_(config_message.tunneling_config().propagate_response_trailers()),
+      post_path_(config_message.tunneling_config().post_path()),
+      route_stat_name_storage_("tcpproxy_tunneling", context.scope().symbolTable()),
+      router_config_(route_stat_name_storage_.statName(), context.localInfo(), stats_scope,
```

Comment:
Just wondering if it would make sense to use the ctor that accepts the Router config proto? I mean, manually craft the config proto and maybe leaving some fields unpopulated such as the http upstream filters / upstream access log / some boolean values that you've set to false here, etc..
Not critical for the PR ofc, just to see if I understand correctly. Thanks

Reply:
thanks @ohadvano . I will try out